### PR TITLE
fix #6252 ([mobile UX] Eur symbol displayed below the price of the item)

### LINKED ISSUE #6252: [mobile UX] Eur symbol displayed below the price of the item
## Description

In the shopfront and when using a mobile device, the Eur currency symbol is displayed below the price of the item. This is observed only for:
• mobile devices
• items with prices in EUR
• prices with 4 digits

See pic below.

## Expected Behavior

Eur currently symbol should be displayed like this  - i.e., after the price:

## Actual Behaviour

Eur currently symbol is displayed like this - i.e., below the price:

## Steps to Reproduce

1. Got to Browserstack and choose a mobile device (tested for Android/iPhone Firefox/Chrome/Safari).
2. Choose a shop with Eur currency and items with price which are at least 4 digits long 
3. Browse through the products and see the bug.

## Animated Gif/Screenshot

Below, left side - staging-UK - pound currency is correctly displayed. Compare with pic on the right side staging-ES, Eur currency below the price (red arrows):

image

## Workaround

Rotating your device 90°.

## Severity

bug-s3: a feature is broken but there is a workaround

## Your Environment

* Version used: v3.3.0
* Browser name and version: Firefox 81 
* Operating System and version (desktop or mobile): Mobile Android

### PR DESCRIPTION
**Pull request by andrewpbrett**
#### What? Why?

Closes #6252 

The formatting string for the currency display put an extra space in between the amount and the currency symbol, which resulted in it wrapping onto a second line. 

#### What should we test?
See the issue; you can change the instance config to put the currency symbol after the amount and a price with four digits should remain all on one line. 

#### Release notes
Fixed a visual bug where the price for an item would split across two lines on mobile devices. 

Changelog Category: User facing changes

### CODE REVIEW

**Review by Matt-Yorkley — APPROVED** [review]

File: app/assets/javascripts/darkswarm/filters/localize_currency.js.coffee [hunk 1]
```diff
@@ -6,7 +6,7 @@ Darkswarm.filter "localizeCurrency", (currencyConfig)->
     # Set decimal points, 2 or 0 if hide_cents.
     decimals = if currencyConfig.hide_cents == "true" then 0 else 2
     # Set format if the currency symbol should come after the number, otherwise (default) use the locale setting.
-    format = if currencyConfig.symbol_position == "after" then "%n %u" else undefined
+    format = if currencyConfig.symbol_position == "after" then "%n%u" else undefined
```
Comment (sauloperez) [inline app/assets/javascripts/darkswarm/filters/localize_currency.js.coffee:9]: I took me a few seconds to realize `%n%u` is a formatting string :sweat_smile:

**Review by sauloperez — APPROVED** [review]

**Comment by filipefurtad0** [comment]
Hey @andrewpbrett ,

I checked this bug again in staging. I've set different price "lengths" in order to mimic other weaker currencies.

- On mobile (Android), it looks like this:

Chrome
image

Firefox
image


- Turns out, this bug is observable as well in desktop as well. If one inserts an item into the cart, a second row is shown, see below:

I took a series of pics, for different screen sizes - **see top right corner of the pic** (Firefox 82, Ubuntu 20.04):

image

image

image

image

image

image

image

I staged this PR, but somehow can't quite observe any changes:

Chrome
image

Firefox
image

This video shows the behavior after staging this PR. Below is Firefox 82, and below Chrome 86:

currency_down_before_

Perhaps this PR could still be improved and extended to other number sizes? Happy to get feedback Andy, in case there is a testcase I'm missing :+1:

**Comment by Matt-Yorkley** [comment]
Maybe we would be better off solving this with CSS? I think there's a few ways to control line-wrapping bahavior with CSS rules.

**Comment by filipefurtad0** [comment]
Thanks for feedback on this one @Matt-Yorkley ,
Moving back to In Dev for now :+1:

**Comment by sauloperez** [comment]
I could take a look myself once I'm done with https://github.com/openfoodfoundation/openfoodnetwork/pull/6366. You guys are pretty busy with many other things and I think this fits in my rather limited OFN hours per week.

**Comment by andrewpbrett** [comment]
I took another look and found a quick solution - sorry @sauloperez  :) 

I tested on my iPhone SE (the original, so 4" screen) and it looks like this should work for up to about 7 digits (depending slightly on what those digits are). Anything larger than that can accommodate even more digits before running over into the add to cart button.

**Comment by RachL** [comment]
@andrewpbrett FYI we are using browserstack to test on various devices and OS: you should have access to the credentials in bitwarden, let me know if it does not work 👍 1:

**Review by sauloperez — APPROVED** [review]

**Comment by filipefurtad0** [comment]
Hey @andrewpbrett ,

checked the following devices, after staging this PR:

**Android 10 Samsung S10, Chrome**

image

**Android 10 Samsung S20, Firefox**

image

**IOS 14 IPhone 12 - Safari**

image

**Desktop Ubuntu 20.04 - Firefox 83 (responsiveness, identical to Chrome 86)**

_euro_resize_after

I also did some regression tests, by setting the Euro sign before the price. Interestingly, I couldn't set the Euro symbol, to appear before the price on mobile - worked in desktop though and it looked good as well. Ready to go!

**Pull request merged**